client/src/pages/HomePage/HomePageContext.tsx

```diff
@@ -35,14 +35,14 @@ export const useHomePageContext = (): UseHomePageContext => {
       setChartData(transactions as TransactionDataPoint[]);
     } catch (error) {
       console.error('Get stats data error', error);
-    } finally {
-      setLoading(false);
     }
   }, []);
 
   useEffect(() => {
-    getStatsData();
-  }, [getStatsData]);
+    if (response) {
+      getStatsData(response);
+    }
+  }, [response, getStatsData]);
```

Comment:
i think we do not need this, just move getStatsData logic into getStats service method